modular_skyrat/modules/syntheticwound/code/datums/wounds/burn.dm

```diff
@@ -0,0 +1,247 @@
+
+/*
+	Burn wounds
+*/
+
+// TODO: well, a lot really, but specifically I want to add potential fusing of clothing/equipment on the affected area, and limb infections, though those may go in body part code
+/datum/wound/synthetic/burn
+	name = "Burn Wound"
+	a_or_from = "from"
+	wound_type = WOUND_BURN
+	processes = TRUE
+	sound_effect = 'sound/effects/wounds/sizzle1.ogg'
+	wound_flags = (FLESH_WOUND | ACCEPTS_GAUZE)
+
+	treatable_by = list(/obj/item/stack/medical/ointment, /obj/item/stack/medical/mesh) // sterilizer and alcohol will require reagent treatments, coming soon
+
+		// Flesh damage vars
+	/// How much damage to our flesh we currently have. Once both this and infestation reach 0, the wound is considered healed
+	var/flesh_damage = 5
+	/// Our current counter for how much flesh regeneration we have stacked from regenerative mesh/synthflesh/whatever, decrements each tick and lowers flesh_damage
+	var/flesh_healing = 0
+
+		// Infestation vars (only for severe and critical)
+	/// How quickly infection breeds on this burn if we don't have disinfectant
+	var/infestation_rate = 0
+	/// Our current level of infection
+	var/infestation = 0
+	/// Our current level of sanitization/anti-infection, from disinfectants/alcohol/UV lights. While positive, totally pauses and slowly reverses infestation effects each tick
+	var/sanitization = 0
+
+	/// Once we reach infestation beyond WOUND_INFESTATION_SEPSIS, we get this many warnings before the limb is completely paralyzed (you'd have to ignore a really bad burn for a really long time for this to happen)
+	var/strikes_to_lose_limb = 3
+
+
+/datum/wound/synthetic/burn/handle_process()
+	. = ..()
+	if(strikes_to_lose_limb == 0) // we've already hit sepsis, nothing more to do
+		if(prob(1))
+			victim.adjustStaminaLoss(10)
+			victim.visible_message("<span class='danger'>The heat damage on the remnants of [victim]'s melted [limb.name] arc violently and creak dangerously!</span>", "<span class='warning'>You feel a sudden drain from your [limb.name]; causing a heavy sense of exhaustion to rush through your circuitry.</span>", vision_distance = COMBAT_MESSAGE_RANGE)
+		return
+
+	// here's the check to see if we're cleared up
+	if(flesh_damage <= 0)
+		to_chat(victim, "<span class='green'>The burns on your [limb.name] have cleared up!</span>")
+		qdel(src)
+		return
+
+	infestation += infestation_rate //Infestation rate is indicative of burns getting worse due to short circuits
+
+	switch(infestation)
+		if(0 to WOUND_INFECTION_MODERATE)
```

Comment:
```suggestion
```

Reply:
Why does this exist if you dont do anything with it and there is also no else statement